project007/webapp/controller/BaseController.js

```diff
@@ -157,6 +157,8 @@ sap.ui.define(
           default:
             if (this._checkFields("groupEditValueProduct")) {
               MessageBox.alert(this.i18n("AlertMessage"));
+            } else if (this._checkDataInTable("groupEditValueProduct")) {
```

Comment:
Methods in a BaseController supposed to be reusable, which means no hardcoded values. You have hardcoded string values here. And also you have to much if else statements inside switch case. Try to change it